rest/config.go

```diff
@@ -382,18 +383,40 @@ func (dbConfig *DbConfig) setup(name string) error {
 	return nil
 }
 
+// httpSSLVerifyError results when loading config from HTTPS endpoint that's using a
+// self-signed certificate, but the unsupported remote_config_tls_skip_verify option
+// is not set. SSL certificate verification is turned on by default.
+type httpSSLVerifyError struct {
+	// cause contains an error that may be helpful in determining the
+	// underlying cause of SSL certificate verification failure.
+	cause error
+}
+
+func (e httpSSLVerifyError) Error() string {
+	return fmt.Sprintf("%v. SSL certificate verification is turned on by default when loading config over HTTPS. "+
+		"Can be overriden by the unsupported \"remote_config_tls_skip_verify\" option", e.cause)
+}
+
 // loadJavaScript loads the JavaScript source from an external file or and HTTP/HTTPS endpoint.
 // If the specified path does not qualify for a valid file or an URI, it returns the input path
 // as-is with the assumption that it is an inline JavaScript source. Returns error if there is
 // any failure in reading the JavaScript file or URI.
-func loadJavaScript(path string) (js string, err error) {
-	rc, err := readFromPath(path)
+func loadJavaScript(path string, insecureSkipVerify bool) (js string, err error) {
+	rc, err := readFromPath(path, insecureSkipVerify)
 	if errors.Is(err, ErrPathNotFound) {
 		// If rc is nil and readFromPath returns no error, treat the
 		// the given path as an inline JavaScript and return it as-is.
 		return path, nil
 	}
 	if err != nil {
+		if !insecureSkipVerify {
+			switch errors.Unwrap(err).(type) {
```

Comment:
Should use errors.Is here - covers the top-level error and any wrapped error.

Reply:
When looking further, the errors.As seems fits best in this case, because the errors.Is function behaves like a comparison to a sentinel error, and the errors.As function behaves like a type assertion.